doc/src/manual/types.md

```diff
@@ -78,7 +70,10 @@ julia> (1+2)::Int
 ```
 
 This allows a type assertion to be attached to any expression in-place.
-
+When the type on the right is concrete,
+the value on the left must have that type as its implementation.
```

Comment:
```suggestion
```

this is true for dispatch purposes, but not true in general in cases like `x::Int = 'a'`

Reply:
```
When the type on the right is concrete,
it must be [`convert`](@ref)able to the type implemented by the value on the left.
```
?

Reply:
This is covered by the rest of the paragraph:

> When appended to a variable on the left-hand side of an assignment, or as part of a `local` declaration, the `::` operator means something a bit different: it declares the variable to always have the specified type, like a type declaration in a statically-typed language such as C. Every value assigned to the variable will be converted to the declared type using [`convert`](@ref):

Reply:
woops, yes